diff --git a/ddbridge/ddbridge-core.c b/ddbridge/ddbridge-core.c
index 60a00f2..a9709b6 100644
--- a/ddbridge/ddbridge-core.c
+++ b/ddbridge/ddbridge-core.c
@@ -926,16 +926,19 @@ static int locked_gate_ctrl(struct dvb_frontend *fe, int enable)
 	return status;
 }
 
-#ifdef CONFIG_DVB_DRXK
+#if IS_ENABLED(CONFIG_DVB_DRXK)
 static int demod_attach_drxk(struct ddb_input *input)
 {
 	struct i2c_adapter *i2c = &input->port->i2c->adap;
 	struct ddb_dvb *dvb = &input->port->dvb[input->nr & 1];
 	struct dvb_frontend *fe;
+	struct drxk_config config;
 
-	fe = dvb->fe = dvb_attach(drxk_attach,
-				  i2c, 0x29 + (input->nr&1),
-				  &dvb->fe2);
+	memset(&config, 0, sizeof(config));
+	config.adr = 0x29 + (input->nr & 1);
+	config.microcode_name = "drxk_a3.mc";
+
+	fe = dvb->fe = dvb_attach(drxk_attach, &config, i2c);
 	if (!fe) {
 		pr_err("No DRXK found!\n");
 		return -ENODEV;
@@ -1486,7 +1489,7 @@ static int dvb_input_attach(struct ddb_input *input)
 		if (tuner_attach_stv6110(input, 1) < 0)
 			return -ENODEV;
 		break;
-#ifdef CONFIG_DVB_DRXK
+#if IS_ENABLED(CONFIG_DVB_DRXK)
 	case DDB_TUNER_DVBCT_TR:
 		if (demod_attach_drxk(input) < 0)
 			return -ENODEV;
