diff --git a/frontends/Makefile b/frontends/Makefile
index 5e00ccf..e1a05ef 100644
--- a/frontends/Makefile
+++ b/frontends/Makefile
@@ -10,7 +10,7 @@ EXTRA_CFLAGS += -DCONFIG_DVB_STV0367DD
 #EXTRA_CFLAGS += -DCONFIG_DVB_TDA18212
 EXTRA_CFLAGS += -DCONFIG_DVB_TDA18212DD
 EXTRA_CFLAGS += -DCONFIG_DVB_CXD2843
-EXTRA_CFLAGS += -DDBVALS
+#EXTRA_CFLAGS += -DDBVALS
 NOSTDINC_FLAGS += -I$(SUBDIRS)/include -I$(SUBDIRS)/dvb-core
 
 drxk-objs := drxk_hard.o 
