diff --git a/ddbridge/ddbridge-core.c b/ddbridge/ddbridge-core.c
index 1f193ae..1ffd886 100644
--- a/ddbridge/ddbridge-core.c
+++ b/ddbridge/ddbridge-core.c
@@ -321,6 +321,7 @@ static void ddb_output_start(struct ddb_output *output)
 		spin_lock_irq(&output->dma->lock);
 		output->dma->cbuf = 0;
 		output->dma->coff = 0;
+		output->dma->stat = 0;
 		ddbwritel(dev, 0, DMA_BUFFER_CONTROL(output->dma->nr));
 	}
 	if (output->port->class == DDB_PORT_MOD)
@@ -395,6 +396,7 @@ static void ddb_input_start(struct ddb_input *input)
 		spin_lock_irq(&input->dma->lock);
 		input->dma->cbuf = 0;
 		input->dma->coff = 0;
+		input->dma->stat = 0;
 		ddbwritel(dev, 0, DMA_BUFFER_CONTROL(input->dma->nr));
 	}
 	ddbwritel(dev, 0, TS_INPUT_CONTROL2(input->nr));
